tests/unit/apps/test_core.py

```diff
@@ -708,6 +716,32 @@ def test_nova_compute_get_enable_scheduler_step(model, units):
     assert app._get_enable_scheduler_step(units_selected) == expected_step
 
 
+@pytest.mark.parametrize(
+    "units",
+    [
+        [f"nova-compute/{unit}" for unit in range(1)],
+        [f"nova-compute/{unit}" for unit in range(2)],
+        [f"nova-compute/{unit}" for unit in range(3)],
+    ],
+)
+def test_nova_compute_get_restart_subordinate_services_steps(model, units):
+    app = _generate_nova_compute_app(model)
+    units_selected = [app.units[unit] for unit in units]
+    assert app._get_restart_subordinate_services_steps(units_selected) == [
+        PostUpgradeStep(
+            description=f"Restart subordinate service for unit: '{unit.subordinates[0].name}'",
+            coro=model.run_on_unit(
+                unit_name=unit.subordinates[0].name,
```

Comment:
Here I would like to keep it because it's generate from the parametrize input.
The testing logic include we want to make sure all the steps will be generate for every units.